_posts/2016-04-14-Rust-1.8.md

```diff
@@ -0,0 +1,251 @@
+
+### What's in 1.8 stable
+
+There are two new features in Rust 1.8, as well as good news for Windows users!
+Additionally, work is underway to replace our `make`-based build system with
+one based on Cargo.
+
+The first feature is that the various “operator equals” operators, such as `+=`
+and `-=`, are now overloadable via various traits. This change was accepted in
+[RFC 953], and looks like this:
+
+```rust
+use std::ops:: AddAssign;
+
+#[derive(Debug)]
+struct Count { 
+    value: i32,
+}
+    
+impl AddAssign for Count {
+    fn add_assign(&mut self, other: Count) {
+        self.value += other.value;
+    }
+}   
+
+fn main() {
+    let mut c1 = Count { value: 1 };
+    let c2 = Count { value: 5 };
+
+    c1 += c2;
+
+    println!("{:?}", c1);
+}
+```
+
+[RFC 953]: https://github.com/rust-lang/rfcs/blob/master/text/0953-op-assign.md
+
+This will print out `Count { value: 6 }`. Like the other operator traits, an
+associated type allows you to use different types on each side of the operator,
+as well. See the RFC for more details.
+
+The second feature is very small, and comes from [RFC 218]. Before Rust 1.8, a
+`struct` with no fields did not have curly braces:
+
+```rust
+struct Foo; // works
+struct Bar { }; // error
+```
+
+[RFC 218]: https://github.com/rust-lang/rfcs/blob/master/text/0218-empty-struct-with-braces.md
+
+The second form is no longer an error, and is equivalent to the first. This was
+originally disallowed for consistency with other empty declarations, as well as
+a parsing ambiguity. However, that ambiguity is non-existent in post-1.0 Rust,
+and macro authors saw additional complexity due to needing a special-case. Also,
+users who do active development would sometimes switch between empty and
+non-empty versions of a struct, and the extra work and diffs involved was less
+than ideal. Allowing both forms is slightly less consistent, but the ergonomics
+of these two cases was deemed worth it.
+
+On the Windows front, 32-bit MSVC builds [now implement unwinding]. This moves
+`i686-pc-windows-msvc` to a Tier 1 platform.
+
+[now implement unwinding]: https://github.com/rust-lang/rust/pull/30448
+
+Finally, we have used `make` to build Rust for a very long time. However,
+we already have a wonderful tool for building Rust programs: Cargo. In Rust
+1.8, [initial support landed] for a new build system that’s written in Rust,
+and based on Cargo. It is not yet the default, and there is much more work to
+do. We will talk about this in release notes more once it’s completely done,
+for now, please read the GitHub issue for more details.
+
+[initial support landed]: https://github.com/rust-lang/rust/pull/30448
+
+#### Library stabilizations
+
+About 20 library functions and methods are now stable in 1.8. There are three
+major groups of changes: UTF-16 related string methods, various APIs related to
+time, and the various traits needed for operator overloading mentioned in the
```

Comment:
I missed `std::time` the first time I read this, but it's a pretty major feature of this release, could that be highlighted a bit more?